.github/workflows/ci-sanitizers.yml

```diff
@@ -0,0 +1,128 @@
+  # schedule:
+    # - cron: '0 20 * * *' # Every day at 12pm PST (UTC-8)
+  workflow_dispatch:
+
+env:
+  BUILD_TYPE: Debug
+  MOUNT: /azure-osconfig
+  REGISTRY: ghcr.io
+  SANITIZER_FLAGS: -fsanitize=address,undefined -fno-omit-frame-pointer -g -O1
+
+jobs:
+  create-ci-matrix:
+    name: Build CI distro matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-matrix.outputs.matrix }}
+    steps:
+      - name: Get matrix
+        id: get-matrix
+        # Skip centos-7, oraclelinux-7, rhel-7 as they don't currently build properly from Dockerfile
+        run: |
+          matrix="$(cat <<'EOL'
+          [
+            { "name": "ubuntu-22.04", "arch": "amd64", "tag": "latest" }
+          ]
+          EOL
+          )"
+
+          echo Distros to perform CI on: $matrix
+          echo matrix=$matrix >> $GITHUB_OUTPUT
+
+  unit-test:
+    name: Unit test
+    needs: create-ci-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target: ${{ fromJson(needs.create-ci-matrix.outputs.matrix) }}
+          
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          clean: true
+
+      - name: Docker login
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      # - name: Setup QEMU
+      #   if: ${{ matrix.target.arch != 'amd64' }}
+      #   uses: docker/setup-qemu-action@v3
+
+      - name: Run container
+        id: container
+        uses: ./.github/actions/container-run
+        with:
+          registry: ${{ env.REGISTRY }}
+          container: azure/azure-osconfig/${{ matrix.target.name }}-${{ matrix.target.arch }}
+          mount: ${{ github.workspace }}:${{ env.MOUNT }}
+          tag: ${{ matrix.target.tag }}
+
+      - name: Generate build
+        uses: ./.github/actions/container-exec
+        with:
+          container: ${{ steps.container.outputs.id }}
+          cmd: |
+            mkdir build && cd build
+            cmake ../src -DCMAKE_C_FLAGS="${{ env.SANITIZER_FLAGS }}" -DCMAKE_CXX_FLAGS="${{ env.SANITIZER_FLAGS }}" -DCMAKE_build-type=${{ env.BUILD_TYPE }} -Duse_prov_client=ON -Dhsm_type_symm_key=ON -DCOMPILE_WITH_STRICTNESS=ON -DBUILD_TESTS=ON -DBUILD_SAMPLES=ON -DBUILD_ADAPTERS=ON -Duse_default_uuid=ON
+
+      - name: Build azure-osconfig
+        uses: ./.github/actions/container-exec
+        with:
+          container: ${{ steps.container.outputs.id }}
+          working-directory: ${{ env.MOUNT }}/build
+          cmd: cmake --build . --config ${{ env.BUILD_TYPE }} --parallel
+
+      - name: Run ctest
+        uses: ./.github/actions/container-exec
+        with:
+          container: ${{ steps.container.outputs.id }}
+          working-directory: ${{ env.MOUNT }}/build
+          cmd: ctest --verbose > ../${{ matrix.target.name }}-${{ matrix.target.arch }}-sanitizer.log
```

Comment:
>cmd: ctest --verbose [](http://example.com/codeflow?start=10&length=20)

This invokes the unit-tests only, correct? Would be possible to also invoke for the module test recipe functional tests? Not necessarily in this PR, possibly later.

Reply:
Yes, module tests are not yet run by this workflow, planned for later when we see how this one works for us.

Reply:
Sounds good.